bundles/org.openhab.binding.globalcache/src/main/java/org/openhab/binding/globalcache/internal/discovery/MulticastListener.java

```diff
@@ -59,18 +63,17 @@ public class MulticastListener {
     public static final int DEFAULT_SOCKET_TIMEOUT = 3000;
 
     /*
-     * Constructor joins the multicast group, throws IOException on failure.
+     * Constructor joins the multicast group
      */
-    public MulticastListener(String ipv4Address) throws IOException, SocketException {
+    public MulticastListener(String ipv4Address) throws IOException, SocketException, UnknownHostException {
         InetAddress ifAddress = InetAddress.getByName(ipv4Address);
-        NetworkInterface netIF = NetworkInterface.getByInetAddress(ifAddress);
+        NetworkInterface netIF = getMulticastInterface(ipv4Address);
         logger.debug("Discovery job using address {} on network interface {}", ifAddress.getHostAddress(),
-                netIF != null ? netIF.getName() : "UNKNOWN");
+                netIF.getName());
         socket = new MulticastSocket(GC_MULTICAST_PORT);
-        socket.setInterface(ifAddress);
+        socket.setNetworkInterface(netIF);
         socket.setSoTimeout(DEFAULT_SOCKET_TIMEOUT);
-        InetAddress mcastAddress = InetAddress.getByName(GC_MULTICAST_GROUP);
-        socket.joinGroup(mcastAddress);
+        socket.joinGroup(new InetSocketAddress(InetAddress.getByName(GC_MULTICAST_GROUP), GC_MULTICAST_PORT), null);
```

Comment:
No `leaveGroup` needed, for example in method below, `shutdown`?

Reply:
Yes, I think that would make sense.